### DISCUSSION TITLE
How to configure Garnet to use Master-Master replication?

### DISCUSSION THREAD

**Discussion by Degot** [microsoft/garnet#573] [Q&A] [answered]
How to configure next topology?:
node1 - master for slots [ 0 - 8192] slave for slots: [8192 - 16384]
node 2 - master for slots [8192 - 16384] slave for slots: [ 0 - 8192]

It seems that the only way to achieve this is to run two separate instances of GarnetServer on each node. instace1 on node1 - master, instance2 on nod1 - slave... And manage them manually.
Am I right?
Or maybe it's faster to "manually" transfer  Tsavorite.logs between masters only ?

**Comment by vazois** [comment, marked as answer]
This is not possible to do so.  A primary cannot be a replica for a range of slots and a primary for another range.
By default, when setting up a cluster with replication enabled, a given primary will be assigned a range of slots serving read and write requests. 
Its corresponding replica will become an exact copy of that primary serving by default only read requests.
You can technically use the READWRITE command to allow replicas to server reads but this is not recommended and maybe unsafe.
